src/modules/InstURL.rb

```diff
@@ -86,24 +87,27 @@ def RewriteCDUrl(url)
     def installInf2Url(extra_dir = "")
       return @installInf2Url unless @installInf2Url.nil?
 
-      parts = URL.Parse(Linuxrc.InstallInf("ZyppRepoURL"))
+      zypp_repo_url = Linuxrc.InstallInf("ZyppRepoURL").to_s
 
-      if parts.empty?
+      if zypp_repo_url.empty?
         # Make it compatible with the current behaviour when
         # install.inf does not exist.
         log.warn "No URL specified through ZyppRepoURL"
-        parts = URL.Parse("cd:///")
+        zypp_repo_url = "cd:/"
       end
 
-      parts["path"] = File.join(parts["path"], extra_dir) unless extra_dir.empty?
+      uri = URI(zypp_repo_url)
```

Comment:
From my POV, that would make code harder to read and extend. Now we have two different cases to modify the URL:

* Adding an `extra_dir`.
* Skipping SSL verification.

If we add another case in the future, you'll need to parse the URL a third time. From my POV, the current implementation is more straightforward. Anyway, I see Martin's point.

So I'm fine with changing it to something like (names may change):

```ruby
zypp_repo_url = add_extra_dir(zypp_repo_url, extra_dir)
zypp_repo_url = add_ssl_verify(zypp_repo_url)
```